### PR TITLE
Adjust to new Broker pallet.

### PR DESCRIPTION
**Pull request by eskimor**
- [ ] Does not require a CHANGELOG entry

### CODE REVIEW

**Comment by muharem** [comment]
@eskimor is it still draft or ready for review? also some required CIs are falling

**Comment by eskimor** [comment]
@muharem  so far draft. The polkadot-sdk release this PR relies on does not exist yet.

**Comment by seadanda** [comment]
If this enacts before `rotate_sale` and at least one core is sold on the open market we start at 230 KSM, decreasing with a step to a minimum of 2.3.

If no cores are sold it will be 2300 down to 23 and we'll lose a month of info about the new pricing model. Currently we're in the fixed price stage and sellout price is still None, so that's quite likely IMO.

If it enacts after `rotate_sale` then it will be adapted by the old price adapter relative to the number of cores sold. currently we've sold 7/15 (renewals) and we're at the fixed price, so the base price would be about 12.8 KSM. That means the price varies between 1280 and 12.8. But in this case the new renewals coming off a lease will have a high price IIUC.

After weighing that up I would say a migration for the base price makes sense, but only if we are sure we can enact and perform the runtime upgrade in time before the sale is rotated, otherwise I would say we schedule it for the interlude and add a migration for the renewal prices for those coming off their leases the next month who have high prices since 12 KSM isn't an awful minimum. WDYT?

**Comment by seadanda** [comment]
> But the same would be true if no core would be sold as we then take the [current base price as target price](https://github.com/paritytech/polkadot-sdk/blob/f4dc8d22b49866a10fb720acee6a2c0e3249e22b/substrate/frame/broker/src/tick_impls.rs#L231).

Ah great, knew I was missing something - this makes more sense. I was mixing up the target price and base/min/regular/fixed price (🫠).

> So yeah, a migration adjusting the renewal price to the current price makes sense. We should also adjust periods - the fix price length can be much shorter.

I agree. However in general this is the minimum time that the secondary markets have to operate within, so we don't want to go too short on it. Maybe roughly 1:1 primary:secondary? Or what were you thinking? I suppose it could even be 2:1 if we expect the target price to be usually correct long term.

**Review by seadanda — APPROVED** [review]

**Comment by eskimor** [comment]
@muharem ready for review.

File: system-parachains/coretime/coretime-kusama/src/coretime.rs [hunk 1]
```diff
@@ -253,8 +214,5 @@ impl pallet_broker::Config for Runtime {
 	type WeightInfo = weights::pallet_broker::WeightInfo<Runtime>;
 	type PalletId = BrokerPalletId;
 	type AdminOrigin = EnsureRoot<AccountId>;
-	#[cfg(feature = "runtime-benchmarks")]
-	type PriceAdapter = pallet_broker::Linear;
-	#[cfg(not(feature = "runtime-benchmarks"))]
-	type PriceAdapter = PriceAdapter;
```
Comment (ordian) [inline system-parachains/coretime/coretime-kusama/src/coretime.rs:259]: for my education: are benchmarks updated in a separate PR? what is the current process? #324?

**Review by ordian — APPROVED** [review]

**Comment by eskimor** [comment]
bot merge

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**